### PR TITLE
feat: add v2 frontend UI design

### PR DESCRIPTION
**Pull request by kittybest**
**Description**

- [x] modify homepage
- [x] add round information
- [x] remove unused environment variables
- [x] read `roundId` from website url instead of adding to the page props
- [x] rearrange the folder structure, since the projects/ballot should be inside the rounds

**Notice**
This branch would be merged to `dev` instead of `main` since the round data are fake data now, just for frontend UI/UX testing, we should get the round data from the `registry` contract later.

### CODE REVIEW

File: packages/interface/src/components/Header.tsx [hunk 1]
```diff
@@ -85,12 +86,20 @@ const Header = ({ navLinks }: IHeaderProps) => {
...
+            const pageLen = asPath.split("/").length;
+
             return (
-              <NavLink key={link.href} href={link.href} isActive={asPath.startsWith(pageName)}>
+              <NavLink
+                key={link.href}
+                href={link.href}
+                isActive={
+                  (pageLen === 3 && link.children === "Projects") ||
+                  (pageLen > 3 && asPath.split("/")[3] === link.href.split("/")[3])
```
Comment (0xmad) [inline packages/interface/src/components/Header.tsx:97]: There are magic numbers

File: packages/interface/src/contexts/Round.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  const getRound = (roundId: string): Round | undefined => rounds.find((round) => round.roundId === roundId);
+
+  const value = useMemo(
+    () => ({
+      rounds,
+      getRound,
+    }),
+    [getRound],
```
Comment (0xmad) [inline packages/interface/src/contexts/Round.tsx:27]:
```suggestion
    [rounds, getRound],
```

File: packages/interface/src/contexts/Round.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      roundId: "open-rpgf-1",
+      description: "This is the description of this round, please add your own description.",
+      startsAt: 1723477832000,
+      registrationEndsAt: 1723487832000,
+      votingEndsAt: 1724009826000,
+      tallyURL: "https://upblxu2duoxmkobt.public.blob.vercel-storage.com/tally.json",
+    },
+  ];
+
+  const getRound = (roundId: string): Round | undefined => rounds.find((round) => round.roundId === roundId);
```
Comment (0xmad) [inline packages/interface/src/contexts/Round.tsx:20]: You need to use `useCallback` here

File: packages/interface/src/contexts/Round.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+export const RoundProvider: React.FC<RoundProviderProps> = ({ children }: RoundProviderProps) => {
+  const rounds = [
+    {
+      roundId: "open-rpgf-1",
+      description: "This is the description of this round, please add your own description.",
+      startsAt: 1723477832000,
+      registrationEndsAt: 1723487832000,
+      votingEndsAt: 1724009826000,
+      tallyURL: "https://upblxu2duoxmkobt.public.blob.vercel-storage.com/tally.json",
+    },
```
Comment (0xmad) [inline packages/interface/src/contexts/Round.tsx:17]: What's the reason to keep these config fields here?

File: packages/interface/src/pages/index.tsx [hunk 5]
```diff
@@ -1,15 +1,97 @@
...
+const HomePage = (): JSX.Element => {
+  const { isConnected } = useAccount();
+  const { isRegistered } = useMaci();
+  const isAdmin = useIsAdmin();
+  const { rounds } = useRound();
 
-export default SignupPage;
+  return (
+    <Layout type="home">
+      {/* <EligibilityDialog />
```
Comment (0xmad) [inline packages/interface/src/pages/index.tsx:22]: Please, remove commented code

File: packages/interface/src/providers/index.tsx [hunk 6]
```diff
@@ -37,11 +38,13 @@ export const Providers = ({ children }: PropsWithChildren): JSX.Element => {
       <WagmiProvider config={config}>
         <QueryClientProvider client={queryClient}>
           <RainbowKitProvider theme={customTheme}>
-            <MaciProvider>
-              <BallotProvider>{children}</BallotProvider>
+            <RoundProvider>
+              <MaciProvider>
+                <BallotProvider>{children}</BallotProvider>
```
Comment (0xmad) [inline packages/interface/src/providers/index.tsx:43]: It's time to add state management. There are already three providers.

**Review by 0xmad — CHANGES_REQUESTED** [review]
@kittybest thanks, there are some comments

File: packages/interface/src/contexts/Round.tsx (hunk 4 above)
Comment (kittybest) [reply packages/interface/src/contexts/Round.tsx:17]: just as I mentioned in the description, these data should be got from the registry contract, now I'm testing out the frontend components, so just put fake data here.

File: packages/interface/src/providers/index.tsx (hunk 6 above)
Comment (kittybest) [reply packages/interface/src/providers/index.tsx:43]: are you saying to reduce the providers to single provider, or something else?

File: packages/interface/src/contexts/Round.tsx (hunk 4 above)
Comment (0xmad) [reply packages/interface/src/contexts/Round.tsx:17]: This data will be available using metadataUrl from registry contract. Registry won't have this data.

File: packages/interface/src/providers/index.tsx (hunk 6 above)
Comment (0xmad) [reply packages/interface/src/providers/index.tsx:43]: I'm saying about state management (redux, effector, mobx and etc)

File: packages/interface/src/contexts/Round.tsx (hunk 4 above)
Comment (kittybest) [reply packages/interface/src/contexts/Round.tsx:17]: yes but still need these fake data to test the frontend? or you suggest to upload these data as json file to vercel and add the URL in the env variable?

File: packages/interface/src/providers/index.tsx (hunk 6 above)
Comment (kittybest) [reply packages/interface/src/providers/index.tsx:43]: what would you recommend for our project?

Comment (0xmad) [reply packages/interface/src/providers/index.tsx:43]: I prefer effector but redux is a standard. So it's up to you what to choose.

File: packages/interface/src/contexts/Round.tsx (hunk 4 above)
Comment (kittybest) [reply packages/interface/src/contexts/Round.tsx:17]: yes sure, it should be a service to fetch these metadata from the metadataURL provided by registry contract, and the registry contract address is provided inside the env file.
The metadataURL is added once the user start a new round --> so once a user setup a new round and click deploy, the data are uploaded to a certain storage, and a url is added to the registry contract.

I think this is the full process of the round information, and it shouldn't be included in this PR, that's why I use fake data now.

File: packages/interface/src/providers/index.tsx (hunk 6 above)
Comment (kittybest) [reply packages/interface/src/providers/index.tsx:43]: Haven't used effector before, only used mobx and redux, let me take a look at the effector stuff and decide.

File: packages/interface/src/contexts/Round.tsx (hunk 4 above)
Comment (0xmad) [reply packages/interface/src/contexts/Round.tsx:17]: Sure, for now let's keep fake data.

**Review by 0xmad — APPROVED** [review]
@kittybest thanks!

File: packages/interface/src/providers/index.tsx (hunk 6 above)
Comment (kittybest) [reply packages/interface/src/providers/index.tsx:43]: open an issue here: #272

**Pull request merged**